var/spack/repos/builtin/packages/cscope/package.py

```diff
@@ -33,5 +33,12 @@ class Cscope(AutotoolsPackage):
 
     version('15.8b', '8f9409a238ee313a96f9f87fe0f3b176')
 
-    # Can be configured to use flex (not necessary)
-    # ./configure --with-flex
+    variant('flex', default=False, description="Builds with flex")
+
+    depends_on('flex', when='+flex')
+
+    def configure_args(self):
+        args = []
+        if 'flex' in self.spec:
```

Comment:
not equivalent -- one checks the variant called flex and one checks the presence of a dependency.  If the variant controls an optional dependency, and the dependency is present when the variant is `True`, then they are equivalent.

If you have A that optionally depends on B that optionally depends on C, you can check for C in A's build with the first one but not the second one.